### PR TITLE
Make needed changes and improvements to start esgf-dashboard-ip

### LINKED ISSUE #615: esgf-dashboard-ip not starting
## LD_LIBRARY_PATH
Using 
```
/usr/local/esgf-dashboard-ip/bin/ip.service start
```
Reports the following error
```
/usr/local/esgf-dashboard-ip/bin/esgf-dashboard-ip: error while loading shared libraries: liblzma.so.5: cannot open shared object file: No such file or directory
```
`liblzma.so.5` is intended to be found in `/usr/local/conda/envs/esgf-pub/lib`, but it is not included in the `LD_LIBRARY_PATH` in `/etc/esg.env`. `ip.service` has the following line.
```
[ -e /etc/esg.env ] && $(grep LD_LIBRARY_PATH /etc/esg.env)
```
This overwrites any manual attempt to specify the `LD_LIBRARY_PATH` with the specification in `/etc/esg.env`.
This environment variable should include the conda env lib path above.
## Missing Properties
Additionally, after manually resolving the above error, the following error is reported by `ip.service start`.
```
[START] ESGF_HOME attribute not found... setting /esg as default
[START] ESGF_HOME = [/esg/]
[START] //esg//config/esgf.properties
[START] Debug level 2 (1=ERROR, 2=WARNING, 3=DEBUG)
***************************************************
[ERROR][esgf-dashboard-ip.c][567] Please note that 9 DB properties are missing in the esgf.properties file. Please check! Exit
[ERROR][esgf-dashboard-ip.c][568] Mandatory properties are:
[ERROR][esgf-dashboard-ip.c][569] [db.host], [db.database], [db.port], [db.user]
[ERROR][esgf-dashboard-ip.c][570] [esgf.host]
[ERROR][esgf-dashboard-ip.c][572] [esgf.registration.xml.path]
[ERROR][esgf-dashboard-ip.c][573] [esgf.registration.xml.download.url]
[ERROR][esgf-dashboard-ip.c][574] [dashboard.ip.app.home]
[ERROR][esgf-dashboard-ip.c][575] Please check!!!
```
My initial thoughts are that the `9 DB properties are missing` is misleading. Looking in `/esg/config/esgf.properties` it looks like only the following are missing
```
esgf.registration.xml.path
esgf.registration.xml.download.url
dashboard.ip.app.home
```
I do not know what these values should be.

### PR DESCRIPTION
**Pull request by nathanlcarlson**
This PR has number of changes and improvements that not only make it more possible to start esgf-dashboard-ip, but also will be useful to other parts of the installer. The goal is to resolve #615 and #462.

### CODE REVIEW

**Comment by nathanlcarlson** [comment]
It is unclear yet if the dashboard is in "working" order. It does appear to be starting, with some warnings, but it is not completely out of order. This is no longer a WIP.

File: esgf_utilities/esg_property_manager.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import logging
 import ConfigParser
 import yaml
-from configobj import ConfigObj
+from backports import configparser
```
Comment (William-Hill) [inline esgf_utilities/esg_property_manager.py]: Did you confirm that the backported configparser maintains comments when writing to the properties file?  That was the main reason for using ConfigObj.  I do have an issue open for creating a guide for the options of the esgf.properties file but was using comments as an extra layer of instruction for users.

Comment (nathanlcarlson) [reply esgf_utilities/esg_property_manager.py]: It does not preserve them in the `/esg/config/esgf.properties` file. Do they need to be there?

Comment (William-Hill) [reply esgf_utilities/esg_property_manager.py]: I was going to use the comments as a way to provide info on some of the more vague configurable options initially.  But I have this issue (#418) open that I plan to add to the documentation page.  That will supersede the comments in esgf.properties. So in short, your approach should be fine when I finish #418

**Pull request merged**